internal/engine/integration/procedure_test.go

```diff
@@ -264,6 +264,35 @@ func Test_Procedures(t *testing.T) {
 				}
 			}`,
 		},
+		{
+			name: "arrays",
+			// all arrays are 1-indexed
+			procedure: `procedure array_funcs() public view {
+				$arr int[] := [1, 2, 3];
+				$arr := array_append($arr, 4);
+				if $arr != [1, 2, 3, 4] {
+					error('array_append failed');
+				}
+
+				$arr2 := $arr[2:4]; // should be [2, 3, 4]
+				if $arr2 != [2, 3, 4] {
+					error('array slice failed');
+				}
+
+				$arr3 := array_prepend(0, $arr);
+				if $arr3 != [0, 1, 2, 3, 4] {
+					error('array_prepend failed');
+				}
+
+				if array_remove($arr3, 3) != [0, 1, 2, 4] {
```

Comment:
@Yaiba the test is here

Reply:
Ok, so this test is against acutal postges.

Not a big deal now, but what about some tests on `PGFormat`?

Reply:
This tests PGFormat.

PGFormat generates Postgres code for the function. The best way to test this IMO is to actually run the generated code against Postgres (which we are doing)

Reply:
hmm, this is not obvious to me since we changed the output of some function's PGFormat in this PR, but no tests are changed.  I guess I'm talking about unit test.

Reply:
I guess it's not a big issue, we do have this 'integration' tests